src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -437,12 +414,11 @@ public DateTime Add(TimeSpan value)
         // time units to this DateTime.
         private DateTime Add(double value, int scale)
         {
-            double millis_double = value * (double)scale + (value >= 0 ? 0.5 : -0.5);
-
-            if (millis_double <= (double)-MaxMillis || millis_double >= (double)MaxMillis)
-                throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_AddValue);
-
+            double millis_double = value * scale + (value >= 0 ? 0.5 : -0.5);
+            if (millis_double <= -MaxMillis || millis_double >= MaxMillis) ThrowOutOfRange();
             return AddTicks((long)millis_double * TicksPerMillisecond);
+
+            static void ThrowOutOfRange() => throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_AddValue);
```

Comment:
>ThrowOutOfRange [](start = 24, length = 15)

don't we have any other helper method doing the same instead of defining this one here?

Reply:
There's only one other identical use for this in `Calendar` (+1 with different parameter which doesn't currently have a `ExceptionArgument` enum value) - should I move it to `ThrowHelper`?

Reply:
I prefer moving it to the helper but would be ok you think otherwise too.

Reply:
In light of https://github.com/dotnet/runtime/issues/46940#issuecomment-759923313 I'd leave it just here for now.

Reply:
that is fine. Thanks!